README.md

```diff
@@ -1,2 +1,189 @@
+### 1. Create consumer
+> Endpoint: `/`  
+> Method: `POST`  
+> Request payload: `ConsumerRequest`  
+
+Creates a new consumer using the provided details.
+
+### 2. Get consumer
+> Endpoint: `/<consumerId>`  
+> Method: `GET`  
+
+Retrieves the details of the consumer with the given ID. 
+
+### 3. Delete consumer
+> Endpoint: `/<consumerId>`  
+> Method: `DELETE`  
+
+Deleted the consumer with the given ID. Returns a 404 Not Found error if a consumer with the given ID is not found. 
+
+### 4. Update consumer
+> Endpoint: `/<consumerId>`  
+> Method: `POST`  
+> Request payload: `ConsumerRequest`  
+
+Updates the details of the consumer with the given ID using the provided details.
+
+These four endpoints showcases the basic CRUD functionalities and how it can be achieved with Ballerina.
+
+### Data storage and retrieval
+The consumer service has it's own MySQL database for storing relevant consumer data. Since this service does not access data outside of it's own module, there is no requirement to make any REST API calls to access the other microservices.
+
+### Order validation
+This service also provides the endpoint `<consumerId>/validate` to validate an order placed by a consumer. This method is currently a dummy method, and does not perform any functional business logic. 
+
+### Running the module
+1. Set up a MySQL database and create the relevant tables using the queries in the `init.sql` file in the `consumer` directory.
+2. Configure the database connection properties in the `Config.toml` file in the `consumer` directory. 
+3. To run the consumer service, simply navigate to the `consumer` directory and execute `bal run` in the terminal. 
+
+## Restaurant service
+The restaurant service represents a restaurant, with its menu, menu items and prices. A restaurant can contain multiple menus; and a menu can contain multiple menu items.
+
+The basic data structure of a restaurant and its associated data types are as follows.
+```ballerina
+type Restaurant record {|
+    int id;
+    string name;
+    string address;
+    Menu[] menus;
+|};
+
+type Menu record {|
+    int id;
+    string name;
+    MenuItem[] items;
+|};
+
+type MenuItem record {|
+    int id;
+    string name;
+    decimal price;
+|};
+```
+
+This service provides endpoints to perform basic CRUD functionalities with restaurants, menus and menu items through 12 different endpoints.
+
+### Managing tickets
+The restaurant service also provides endpoints to create and view tickets, as well as update their status. A ticket contains the following information:
+```ballerina
+type Ticket record {|
+    int id;
+    Restaurant restaurant;
+    Order 'order;
+    TicketState status;
+|};
+```
+
+A ticket can be in one of the following states:
+* `ACCEPTED`
+* `PREPARING`
+* `READY_FOR_PICKUP`
+* `PICKED_UP`
+
+When a ticket is created, it will initially be in the `ACCEPTED` state. The state of a ticket can be advanced to each state using the three endpoints defined.
```

Comment:
```suggestion
When a ticket is created, it will initially be in the `ACCEPTED` state. The state of a ticket can be advanced to each state below using the three endpoints defined.
```